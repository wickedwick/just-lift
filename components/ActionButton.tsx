import React from 'react'
import { Button } from 'react-native-paper'
import { ActionButtonProps } from '../types/common'

const ActionButton = (props: ActionButtonProps): JSX.Element => {
  return (
    <Button
      icon={props.icon}
      mode={props.contained ? 'contained' : 'text'}
      style={props.style}
      onPress={props.onPress}
    >
      {props.text}
    </Button>
  )
}

export default ActionButton
