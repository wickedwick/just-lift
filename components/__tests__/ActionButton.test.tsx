import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import { Button } from 'react-native-paper'
import ActionButton from '../ActionButton'

configure({ adapter: new Adapter() })

const props = {
  icon: 'add',
  text: 'Add',
  onPress: jest.fn(),
  style: {},
}

describe('<ActionButton />', () => {
  it('should render and match the snapshot', () => {
    const wrapper = shallow(<ActionButton {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('calls the onPress function on button press', () => {
    const wrapper = shallow(<ActionButton {...props} />)
    wrapper.find(Button).simulate('press')
    expect(props.onPress).toHaveBeenCalled()
  })
})