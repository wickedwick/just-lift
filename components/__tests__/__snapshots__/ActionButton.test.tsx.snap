// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionButton /> should render and match the snapshot 1`] = `ShallowWrapper {}`;
